docs/DesignDocs/CustomAttestation_V3.md

```diff
@@ -0,0 +1,423 @@
+Custom Attestation Data Formats for Open Enclave V3 Updates
+================
+
+As custom attestation plugins are developed and features are requested, new APIs were added to:
+
+1. Query built-in SGX attesters and verifiers to be registered.
+2. Select a format of the evidence that the registered attesters can generate.
+3. Query evidence parameters to be used as optional inputs when generating evidences.
+
+The following API was affected:
+
+1. Built-in SGX attesters and verifiers
```

Comment:
This is a replacement of existing SGX-specific API.
Old:
- oe_sgx_plugin_attester
- oe_sgx_plugin_verifier
New:
- oe_sgx_get_attesters
- oe_sgx_get_verifiers

---
In reply to: [393244871](https://github.com/openenclave/openenclave/pull/2621#discussion_r393244871) [](ancestors = 393244871)

Reply:
That doesn't actually answer my real question.  My question is why the new replacements are sgx specific instead of agnostic like say "oe_get_attester_plugins", and "oe_get_verifier_plugins".  Offhand, I would recommend the latter names.  My general philosophy is to not have any SGX-specific API when a TEE-agnostic API could do the same task.

Reply:
Renamed
. oe_sgx_get_attesters to oe_get_attesters
. oe_sgx_get_verifiers to oe_get_verifiers

'plugin' to us is an external/internal unit, typically a library, the includes one or more attesters or verifiers. That's why existing APIs do not have 'plugin' in their names (except the ones to be deprecated). That is also why 'oe_get_attesters' is not 'oe_get_attester_plugins'.

---
In reply to: [394638999](https://github.com/openenclave/openenclave/pull/2621#discussion_r394638999) [](ancestors = 394638999)

Reply:
I am not sure these names are consistent with the terminology in draft-ietf-rats-architecture.  See figure 8 for example.  A verifier can support multiple formats.  "get_verifiers" would imply it can return multiple remote servers that you can talk to, as opposed to multiple formats that you as a verifier server can utilize.   Using "verifier_plugins" or some similar name would address this because then verifier is singular and what's plural is the thing that understands a format.